### ISSUE TITLE
requesting network membership stumbles without feedback

### ISSUE THREAD

**Issue by zuojihong** [ConsenSys/QuorumNetworkManager#113]
After starting node1 as coordinator node, I start node2 to join the network. However, after spitting out "Requesting network membership. This will block until the other node responds", there's no further feedback either from node2 or node1. 

BTW: The firewall rules I configured include:
- Allow inbound/outbound raft-http 40000
- Allow inbound/outbound communication-node 50000
- Allow inbound/outbound communication-node-rpc 50010
- Allow inbound/outbound communication-node-ws-rpc 50020
- Allow inbound/outbound geth-node 20000
- Allow inbound/outbound geth-node-rpc 20010
- Allow inbound/outbound geth-node-ws-rpc 20020
- Allow inbound/outbound devp2p 30303
- Allow inbound/outbound constellation-network 9000

Procedures for configuring node 1:
----------------------------------------
oocladmin@oocl-node1:~/workspace/QuorumNetworkManager$ node index.js
Trying to get public ip address, please wait a few seconds...
Welcome!

Please enter the IP address other nodes will use to connect to this node.

Also, please enter a publicly identifyable string for this node to use.

prompt: localIpAddress:  (52.230.126.182)
prompt: nodeName:  oocl-node1
Please select an option:
1) Raft
2) QuorumChain [Disabled - this does not work on newer versions of Quorum]
5) Kill all geth and constellation
prompt: option:  1
Please select an option below:
----- Option 1 and 2 are for the initial setup of a raft network -----
1) Start a node as the setup coordinator [Ideally there should only be one coordinator]
2) Start a node as a non-coordinator
----- Option 3 is for joining a raft network post initial setup  -----
3) Join a raft network if you were not part of the initial setup
4) TODO: Start whisper services and attach to already running node
5) Kill all geth constellation-node
0) Quit
prompt: option:  1
Please select an option below:
1) Allow anyone to connect
2) [TODO] Allow only people with pre-auth tokens to connect
prompt: option:  1
Please select an option below:
1) Clear all files/configuration and start from scratch[WARNING: this clears everything]
2) Keep old files/configuration intact and start the node + whisper services
prompt: option:  1
[*] Starting new node...
Generating node key

enode: enode://5c91d4ce2ccd2f3e9327ece9f9ee16ebc980cfbfefdcff18a3c3394163f9b518a177133127282d06aad2fa9091297420f81d3c25e1132646a7533c29b46a9657@52.230.126.182:20000?raftport=40000

[*] Starting communication node...
[*] RPC connection established
[*] Node started
Please wait for others to join. Hit any key + enter once done.
prompt: done:

Procedures for configuring node 2:
----------------------------------------
oocladmin@oocl-node2:~/workspace/QuorumNetworkManager$ node index.js
Trying to get public ip address, please wait a few seconds...
Welcome!

Please enter the IP address other nodes will use to connect to this node.

Also, please enter a publicly identifyable string for this node to use.

prompt: localIpAddress:  (52.230.126.202)
prompt: nodeName:  oocl-node2
Please select an option:
1) Raft
2) QuorumChain [Disabled - this does not work on newer versions of Quorum]
5) Kill all geth and constellation
prompt: option:  1
Please select an option below:
----- Option 1 and 2 are for the initial setup of a raft network -----
1) Start a node as the setup coordinator [Ideally there should only be one coordinator]
2) Start a node as a non-coordinator
----- Option 3 is for joining a raft network post initial setup  -----
3) Join a raft network if you were not part of the initial setup
4) TODO: Start whisper services and attach to already running node
5) Kill all geth constellation-node
0) Quit
prompt: option:  2
Please select an option below:
1) Clear all files/configuration and start from scratch[WARNING: this clears everything]
2) Keep old files/configuration intact and start the node + whisper services
prompt: option:  1
In order to join the network, please enter the ip address of the coordinating node
prompt: ipAddress:  52.230.126.182
[*] Starting new node...
Account: 0x91f9585d5e66cb4f90bfa2fda1e5c27bd31a0cb5
Generating node key

enode: enode://0e1ac7474663bbeb6a9ac4da78039c584cf23a079575e9d2e767458d2b1663e7ae19acc223d0c69a63b066dd83f3b2449a64378f0e125f8df8e65322625ff51f@52.230.126.202:20000?raftport=40000

Joining enode: enode://9443bd2c5ccc5978831088755491417fe0c3866537b5e9638bcb6ad34cb9bcc58a9338bb492590ff200a54b43a6a03e4a7e33fa111d0a7f6b7192d1ca050f300@52.230.126.182:50000
[*] Joining communication network...
[*] RPC connection established
[*] Node started
true
[*] Communication network joined
[*] Requesting network membership. This will block until the other node responds

**Comment by coeniebeyers** [comment]
Hi! Does node1 only display 
```
Please wait for others to join. Hit any key + enter once done.
prompt: done:
```
Or does it display anything else? If it doesn't, it seems like there might be a whisper problem. Can you confirm that your firewall rules have been applied on both node1 and node2? I've not limited outbound ports before, only inbound. Could you test with allowing all outbound?

Also the following ports don't need to be opened:
```
Allow inbound/outbound communication-node-rpc 50010
Allow inbound/outbound communication-node-ws-rpc 50020
```
These are only once you want to start talking to your nodes from another machine, and you will want to restrict them to the specific IP(s?) where the API/APP is running:
```
Allow inbound/outbound geth-node-rpc 20010
Allow inbound/outbound geth-node-ws-rpc 20020
```
You could also just open one of 20010 or 20020 depending on your requirements.

**Comment by zuojihong** [comment]
Hi, 
  Thanks for your kind support! I checked my firewall rules and found the rules were defined only at the subnet level and not at the node level. And after combining the firewall rules, it led to denial of the inbound traffic at the node level. I removed the rules at the subnet level and put the rules at the node level. After that, node 1 and node 2 can generate further output, e.g., node 1 can output "oocl-node2 has joined the network".
  But I found some new issues here. After node 2 spitting out "Requesting genesis block config. This will block until the other node is online", I turned to node 1, hitting any key + enter. Afterwards, node 1 showed duplicate options as below again:

  node 1
--------------------
![image](https://user-images.githubusercontent.com/4454650/33820929-c56ffeea-de8c-11e7-8aa6-9efed8bb135c.png)

  And node 2 showed similar duplicate options as below:
  node 2
--------------------
![image](https://user-images.githubusercontent.com/4454650/33820973-fdb2094c-de8c-11e7-8217-7746b2a9463d.png)

  The duplicate options make me a little puzzled as I cannot tell if the network has been setup successfully. Are these output expected for successful connection?

**Comment by coeniebeyers** [comment]
Your nodes should be up and running. If you ran the setup in a screen then hit `Ctrl + a + d` to detach from screen. Next, use `./attachToLocalQuorumNode.sh` to attach to the quorum node from inside the QuorumNetworkManager folder. You can check with admin.peers to see if the other node shows up.

**Comment by zuojihong** [comment]
Thanks! As update, I'm able to join the node 3 to the existing network as well:)
